core/libcamera_app.cpp

```diff
@@ -126,6 +127,32 @@ void LibcameraApp::OpenCamera()
 	// The queue takes over ownership from the post-processor.
 	post_processor_.SetCallback(
 		[this](CompletedRequestPtr &r) { this->msg_queue_.Post(Msg(MsgType::RequestComplete, std::move(r))); });
+
+	if (options_->framerate)
+	{
+		std::unique_ptr<CameraConfiguration> config = camera_->generateConfiguration({ libcamera::StreamRole::Raw });
+		const libcamera::StreamFormats &formats = config->at(0).formats();
+
+		// Suppress log messages when enumerating camera modes.
+		libcamera::logSetLevel("RPI", "ERROR");
+		libcamera::logSetLevel("Camera", "ERROR");
+
+		for (const auto &pix : formats.pixelformats())
+		{
+			for (const auto &size : formats.sizes(pix))
+			{
+				config->at(0).size = size;
+				config->at(0).pixelFormat = pix;
+				config->validate();
+				camera_->configure(config.get());
+				auto fd_ctrl = camera_->controls().find(&controls::FrameDurationLimits);
+				sensor_modes_.emplace_back(size, pix, 1.0e6 / fd_ctrl->second.min().get<int64_t>());
+			}
+		}
+
+		libcamera::logSetLevel("RPI", "INFO");
+		libcamera::logSetLevel("Camera", "INFO");
```

Comment:
It strikes me that this is pretty similar to what `--list-cameras` does, but I expect there's no very convenient way to share the code? Just thinking out loud.

Reply:
Slightly different to the ``--list-cameras`` case.  In that we set the log target to none, so no messages will ever get printed.  Here we still get some error messages if something goes wrong in libcamera.  Maybe I should change the former?

Reply:
If it's not straightforward, I would say to leave it...

Reply:
I'll leave it for now.